### ISSUE TITLE
cuDNN / MPIWrapper: Cuda failure 2 - out of memory on validation pass (Beta 10)

### ISSUE THREAD

**Issue by ghost** [Microsoft/CNTK#1415]
cuDNN runs convnet training stage fine (3 hosts using MPI with minibatch size chosen to maximize single host GPU) but then crashes when tries to do cross-validation at the end of the epoch:
```
[CALL STACK]
    > Microsoft::MSR::CNTK::CudaTimer::  Stop
    - Microsoft::MSR::CNTK::CudaTimer::  Stop
    - Microsoft::MSR::CNTK::CudaTimer::  Stop
    - Microsoft::MSR::CNTK::GPUMatrix<float>::  Resize
    - Microsoft::MSR::CNTK::Matrix<float>::  Resize
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
    - Microsoft::MSR::CNTK::Matrix<float>::  __autoclassinit2
```
Train and validate run fine when minibatch size chosen to maximize the GPU memory use of a single host on non-conv DNN training.  Not clear why this crashes with covnet - as when validation happens, the actual memory used (per GPU) is marginally same.

This seems related #1412

**Comment by cha-zhang** [comment]
@ghost, has this been resolved by the latest 2.0 release?

**Comment by cha-zhang** [comment]
Closing this issue since it's been too old. If the issue remains, please reopen. Thanks!
